### PR TITLE
Use common vars for install directories

### PR DESCRIPTION
**Pull request by h3xx**
This will make sure everything is installed in system-standardized locations. And of course, each one of the CMAKE_INSTALL_* variables are override-able.

[More info on GNUInstallDirs](https://cmake.org/cmake/help/v3.4/module/GNUInstallDirs.html)

### CODE REVIEW

File: src/libswami/CMakeLists.txt [hunk 1]
```diff
@@ -102,7 +102,7 @@ link_directories (
 )
 
 add_definitions (
-    -DLOCALEDIR="${DATA_INSTALL_DIR}/locale"
+    -DLOCALEDIR=\"${CMAKE_INSTALL_LOCALEDIR}\"
```
Comment (h3xx) [inline src/libswami/CMakeLists.txt:105]: Oh, and here's a tiny quoting fix as well.

**Comment by derselbst** [comment]
Looks good to me, but I would first like to set up CI builds. However, I cannot yet tell when I have time do so.

**Comment by derselbst** [comment]
Works fine, thank you! And sry for the delay, setting up CI builds was more complicated than expected.

**Comment by h3xx** [comment]
Yeah, I totally understand wanting to make sure I didn't break anything. CI is great!

**Pull request merged**